test/python/circuit/test_control_flow.py

```diff
@@ -134,6 +135,30 @@ def test_for_loop_range_instantiation(self):
         self.assertEqual(op.params, [indexset, loop_parameter, body])
         self.assertEqual(op.blocks, (body,))
 
+    def test_for_loop_dynamic_range_instantiation(self):
+        """Verify creation and properties of a ForLoopOp using a dynamic range indexset."""
+        body = QuantumCircuit(3, 1)
+        loop_parameter = Parameter("foo")
+        start = expr.Var.new("start", types.Uint(3))
+        stop = expr.Var.new("stop", types.Uint(3))
+        step = 1
+        indexset = DynamicRange(start, stop, step)
```

Comment:
We could do with a couple of more thorough tests that exercise more of the new code.  In particular, the new OQ3 paths aren't tested right now.